src/Microsoft.DotNet.InternalAbstractions/Native/NativeMethods.Windows.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.InternalAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Windows
+        {
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct RTL_OSVERSIONINFOEX
+            {
+                internal uint dwOSVersionInfoSize;
+                internal uint dwMajorVersion;
+                internal uint dwMinorVersion;
+                internal uint dwBuildNumber;
+                internal uint dwPlatformId;
+                [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
+                internal string szCSDVersion;
+            }
+
+            // This call avoids the shimming Windows does to report old versions
+            [DllImport("ntdll")]
+            private static extern int RtlGetVersion(out RTL_OSVERSIONINFOEX lpVersionInformation);
```

Comment:
Can we just use RuntimeInformation instead of all of these DllImports?  If there's a problem with RuntimeInformation, we should fix that so it can be usable in these situations.

cc: @Priya91, @weshaggard